### DISCUSSION TITLE
Just getting started and 'Code completion engine has stopped working'?

### DISCUSSION THREAD

**Discussion by Sprog-Rockatansky** [pybricks/support#1275] [General]
Greetings folks,

Following the steps on the website and at the point where I'm ready to paste copied code into the editor and for some reason 'Code completion engine has stopped working' appears as a ! down the bottom left. I know the code has been copied however the editor space flashes red once when pasting the code and I can't progress.

Any tips for a brand new user?
![01  31 Oct 23 Code completion engine has stopped working](https://github.com/pybricks/support/assets/149492094/79582536-3ce9-4bde-80c4-ad9c18eb85b6)

**Comment by laurensvalk** [comment]
You have to open a file before you can type/paste code:

Click on the + icon to create a new file.

![image](https://github.com/pybricks/support/assets/12326241/986d6174-4187-4997-b034-2e2b4e364e49)

**Comment by Sprog-Rockatansky** [comment]
Thanks laurensvalk,

Would you believe my last resort was to go through the 'Welcome Tour' where it actually showed me that 'Create New Project' opens a fresh page of the editor to paste that pesky code into. Amazingly only after two minutes I managed to get it functioning and drove the tracked 42140 around the house for a good ten minutes for it's first run.
![42140](https://github.com/pybricks/support/assets/149492094/9879424a-ccd5-49cf-bd1e-f11bed1b5d57)


Awesome!

**Reply by laurensvalk** [reply]
Glad it worked!

I have to agree that it isn't super intuitive currently. We have a few ideas here to add an extra button to the empty home screen to quick start a new file: https://github.com/pybricks/support/issues/1264

**Comment by Sprog-Rockatansky** [comment]
Brilliant,

I was only going off the very few youtube videos for the process explained by a few people. Obviously many changes with Pybricks since those videos were created. Purchased the set and the 88010 Remote Control after seeing multiple 42140's being driven around at Bricktober a few weeks ago by happy little kiddies so I knew it must work. Glad I didn't go with infrared and settled on Bluetooth.
![DSC_0072](https://github.com/pybricks/support/assets/149492094/24b02bc9-da6b-4a71-8798-b136a18e2435)
Cheers!

**Comment by Sprog-Rockatansky** [comment]
Okay,

Question Two: Just tried to go for a second drive... How come the code wasn't stored in the hub and I had to send it again to make it function? The videos talk about a 'include current program' option you could check to store it on the hub but can't find it.

Confused now...

**Reply by laurensvalk** [reply]
The program is automatically saved on the hub now (the videos may be outdated, but that's a good thing -- it's way easier now)
- Press once to turn the hub on (it starts blinking)
- Press again to start/stop your program.

Don't forget to turn on the remote as well. If it's not on, the hub can't find it and nothing appears to happen.

**Comment by Sprog-Rockatansky** [comment]
Yes!

Need to press hub's green button twice and the remote's green button once.

Now sorted! Thank you so much.

**Comment by Sprog-Rockatansky** [comment]
Little old me again folks,

Just scored a 42099 and tried to run the programs to control it with the Remote Control but unfortunately it is not working... any tips? For some reason when I copy the two versions of code ('Autonomous Driving' or 'Powered Up Remote Control') to the hub it's led won't 'pulse warmly blue then turn solid blue when completed'.

![Capture](https://github.com/pybricks/support/assets/149492094/696a6d66-a862-435a-bef1-0c467d8d4445)

**Reply by laurensvalk** [reply]
That example program is using some of our newer features.

If you'd like to try it, use https://beta.pybricks.com, and install the firmware from there as well.

Alternatively, wait a week and this version will come to the stable version you're using as well.

**Comment by Sprog-Rockatansky** [comment]
Got it working via the beta,

Drove it from the study to the library and for some reason it 'forgot' it's code. Quickly re-coded and now it has retained it. Any reason why the hub would 'forget'?

![L42099](https://github.com/pybricks/support/assets/149492094/62939af3-abe2-484b-abc6-f68d74257b45)


Many thanks team!

**Reply by laurensvalk** [reply]
It shouldn't have forgotten its code, but who knows :smile:

Maybe when you ran it the second time, the remote wasn't on anymore. If it can't find the remote, it stops the program, so maybe it just appeared as though nothing happened?

If you want to be sure, next time put a few lines of code at the beginning of the program that turns the LED yellow for a second or something, so you can easily see that something did in fact run.
